xtrack/twiss.py

```diff
@@ -770,7 +772,9 @@ def _twiss_open(line, twiss_init,
     recorded_state = line.record_last_track.state[:, i_start:i_stop+1].copy()
     if not _continue_if_lost:
         assert np.all(recorded_state == 1), (
-            'Some test particles were lost during twiss!')
+             'Some test particles were lost during twiss! '
+          + f'(state {np.unique(recorded_state)}, '
+          + f'at element {np.unique(line.record_last_track.at_element[:, i_start:i_stop+1].copy())})')
```

Comment:
Doesn' this display the element index for all particles, including those that are not lost?

Reply:
Indeed, but as it’s wrapped in `np.unique` it is literally only one extra outputted list item. I chose to do so as I didn’t want to slow down the calculation just for error output. But if you want, I can add it.

Reply:
OK with me